include/extractor/guidance/turn_handler.hpp

```diff
@@ -45,9 +47,24 @@ class TurnHandler : public IntersectionHandler
                             Intersection intersection) const override final;
 
   private:
+    struct Fork
+    {
+        Intersection::iterator right;
+        Intersection::iterator left;
+        const std::size_t size;
+        Fork(Intersection::iterator right, Intersection::iterator left);
```

Comment:
Should this work on `Intersection` or `IntersectionView` - or both?

Check the data these types hold and decide.